raster/r.terraflow/unionFind.h

```diff
@@ -126,13 +126,24 @@ inline void unionFind<T>::makeSet(T x)
     if (x >= maxsize) {
         /* reallocate parent */
         cout << "UnionFind::makeSet: reallocate double " << maxsize << "\n";
-        parent = (T *)realloc(parent, 2 * maxsize * sizeof(T));
-        assert(parent);
-        memset(parent + maxsize, 0, maxsize * sizeof(T));
-        /*reallocate rank */
-        rank = (T *)realloc(rank, 2 * maxsize * sizeof(T));
-        assert(rank);
-        memset(rank + maxsize, 0, maxsize * sizeof(T));
+        T *new_parent = (T *)realloc(parent, 2 * maxsize * sizeof(T));
+        if (!new_parent) {
```

Comment:
```c
        if (!new_parent) {
            assert(new_parent && "Memory reallocation failed for parent");
        } 
 ```
Assert `new_parent` after `if (!new_parent) ` doesn’t make much sense, please reconsider this solution.

Reply:
See https://en.cppreference.com/w/cpp/error/assert; note situation if `NDEBUG ` is set…

Reply:
I have removed the assert statement as suggested. Now the code uses fprintf to handle memory reallocation failures. Please check now.